addon/PageScraper.js

```diff
@@ -33,17 +34,30 @@ PageScraper.prototype = {
    * HTML and insert it in the metadata database
    */
   _asyncParseAndSave: Task.async(function*(rawHTML, url) {
+    let event = this._tabTracker.generateEvent();
+    this._tabTracker.handlePerformanceEvent(event, "receivedRawHTML", Date.now());
+    let startTime = Date.now();
+
     let link = yield this._previewProvider.asyncLinkExist(url);
     if (!link) {
       let metadata;
       try {
         metadata = yield this._metadataParser.parseHTMLText(rawHTML, url);
       } catch (e) {
         Cu.reportError(`MetadataParser failed to parse ${url}`);
+        this._tabTracker.handlePerformanceEvent(event, "metadataParseFail", Date.now() - startTime);
+        return;
       }
+      this._tabTracker.handlePerformanceEvent(event, "metadataParseSuccess", Date.now() - startTime);
+
       if (this._shouldSaveMetadata(metadata)) {
+        this._tabTracker.handlePerformanceEvent(event, "validMetadataReceived");
         this._insertMetadata(metadata);
```

Comment:
Add here:

    this._tabTracker.handlePerformanceEvent(event, "metadataParseSuccess", Date.now() - startTime);